starport/services/networkbuilder/networkbuilder.go

```diff
@@ -356,3 +315,75 @@ func (b *Builder) StartChain(ctx context.Context, chainID string, flags []string
 
 	return g.Wait()
 }
+
+// generateGenesis generate the genesis from the launch information in the specified app home
+func generateGenesis(ctx context.Context, appHome string, chainInfo spn.Chain, launchInfo spn.LaunchInformation, chainCmd *chain.Chain) error {
+	// overwrite genesis with initial genesis.
+	initialGenesis, err := ioutil.ReadFile(initialGenesisPath(appHome))
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(genesisPath(appHome), initialGenesis, 755)
+	if err != nil {
+		return err
+	}
+
+	// make sure that Genesis' genesis_time is set to chain's creation time on SPN.
+	cf := confile.New(confile.DefaultJSONEncodingCreator, genesisPath(appHome))
+	var genesis map[string]interface{}
+	if err := cf.Load(&genesis); err != nil {
+		return err
+	}
+	genesis["genesis_time"] = chainInfo.CreatedAt.UTC().Format(time.RFC3339)
+	if err := cf.Save(genesis); err != nil {
+		return err
+	}
+
+	// add the genesis accounts
+	for _, account := range launchInfo.GenesisAccounts {
+		genesisAccount := chain.Account{
+			Address: account.Address.String(),
+			Coins:   account.Coins.String(),
+		}
+
+		if err := chainCmd.AddGenesisAccount(ctx, genesisAccount, appHome); err != nil {
+			return err
+		}
+	}
+
+	// reset gentx directory
+	dir, err := ioutil.ReadDir(filepath.Join(appHome, "config/gentx"))
+	if err != nil {
+		if strings.Contains(err.Error(), "no such file or directory") {
+			// create the gentx folder if it doesn't exist
+			if err := os.Mkdir(filepath.Join(appHome, "config/gentx"), os.ModePerm); err != nil {
```

Comment:
Instead of running `Mkdir` in case of an `no such file or directory` error, we can always run it before `ReadDir`. This will reduce the code complexity here.